enrollment-server-onboarding-provider-innovatrics/src/main/java/com/wultra/app/onboardingserver/provider/innovatrics/InnovatricsApiService.java

```diff
@@ -145,4 +158,247 @@ public void deleteSelfie(final String customerId, final OwnerId ownerId) throws
+        } catch (Exception e) {
+            throw new RemoteCommunicationException("Unexpected error when creating a new customer resource", e);
+        }
+    }
+
+    /**
+     * Create a new document resource assigned to a customer. This resource is used for documents only, not for selfies.
+     * @param customerId id of the customer to assign the resource to.
+     * @param documentType type of document that will be uploaded later.
+     * @param ownerId owner identification.
+     * @return optional of CreateDocumentResponse. Does not contain important details.
+     * @throws RemoteCommunicationException in case of 4xx or 5xx response status code.
+     */
+    public CreateDocumentResponse createDocument(final String customerId, final DocumentType documentType, final OwnerId ownerId) throws RemoteCommunicationException {
+        final String apiPath = "/api/v1/customers/%s/document".formatted(customerId);
+
+        final DocumentClassificationAdvice classificationAdvice = new DocumentClassificationAdvice();
+        classificationAdvice.setTypes(List.of(convertType(documentType)));
+        classificationAdvice.setCountries(configProps.getDocumentVerificationConfiguration().getDocumentCountries());
+        final DocumentAdvice advice = new DocumentAdvice();
+        advice.setClassification(classificationAdvice);
+        final CreateDocumentRequest request = new CreateDocumentRequest();
+        request.setAdvice(advice);
+
+        try {
+            logger.info("Creating new document of type {} for customer {}, {}", documentType, customerId, ownerId);
+            logger.debug("Calling {}, {}", apiPath, request);
+            final ResponseEntity<CreateDocumentResponse> response = restClient.put(apiPath, request, new ParameterizedTypeReference<>() {});
+            logger.info("Got {} for creating document, {}", response.getStatusCode(), ownerId);
+            logger.debug("{} response status code: {}", apiPath, response.getStatusCode());
+            logger.trace("{} response: {}", apiPath, response);
+            return response.getBody();
+        } catch (RestClientException e) {
+            throw new RemoteCommunicationException("REST API call failed when creating a new document resource for customerId=%s, statusCode=%s, responseBody='%s'".formatted(customerId, e.getStatusCode(), e.getResponse()), e);
+        } catch (Exception e) {
+            throw new RemoteCommunicationException("Unexpected error when creating a new document resource for customerId=%s".formatted(customerId), e);
+        }
+    }
+
+    /**
+     * Provide photo of a document page. A document resource must be already assigned to the customer.
+     * @param customerId id of the customer to whom the document should be provided.
+     * @param side specifies side of the document.
+     * @param data image of the page encoded in base64.
```

Comment:
The `data` parameter seems a bit too generic, can we call it e.g. `imageData` or `imageBytes`?